include/deal.II/lac/sparsity_tools.h

```diff
@@ -204,11 +205,11 @@ namespace SparsityTools
    * DoFHandler::locally_owned_dofs_per_processor and @p myrange are
    * locally_relevant_dofs.
    */
-  template <class DSP_t>
-  void distribute_sparsity_pattern(DSP_t &dsp,
+  template <typename DynamicSparsityType>
```

Comment:
I would replace this by ```SparsityType``` because you use that in the various ```make_sparsity_pattern``` calls that also expect a dynamic pattern.